### PR TITLE
[BEAM-626] Changes in AvroCoder serialization so it can serialize in Kryo

### PR DESCRIPTION
**Pull request by lukecwik**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by lukecwik** [comment]
This is a continuation of #941 
R: @dhalperi @aviemzur
CC: @amitsela

File: sdks/java/core/src/test/java/org/apache/beam/sdk/coders/AvroCoderTest.java [hunk 1]
```diff
@@ -172,6 +175,50 @@ public void testTransientFieldInitialization() throws Exception {
...
+   * @throws Exception
+   */
+  @Test
+  public void testKryoSerialization() throws Exception {
+    Pojo value = new Pojo("Hello", 42);
+    AvroCoder<Pojo> coder = AvroCoder.of(Pojo.class);
+
+    //Kryo instantiation
+    Kryo kryo = new Kryo();
+    kryo.setInstantiatorStrategy(new org.objenesis.strategy.StdInstantiatorStrategy());
```
Comment (tgroh) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/coders/AvroCoderTest.java]: why fully qualified?

Comment (lukecwik) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/coders/AvroCoderTest.java]: Originally got this code from another PR, anyways fixed.

**Comment by amitsela** [comment]
We plan on testing this on cluster (where serialization issues are usually noted), but we had a very busy week, so maybe worth waiting.

**Comment by lukecwik** [comment]
@amitsela Noted, will wait for confirmation.

**Comment by aviemzur** [comment]
👍 
Ran test with a pipeline using spark runner which fails on master and passes using this branch.

**Pull request closed without merging**